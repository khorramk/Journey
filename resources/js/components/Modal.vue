<template>
    <div class="modal bg-blue-300 relative">
        <button class="cross-btn">
            ( X )
        </button>
        <form @submit.prevent="submit">
            <textarea class="text-body" name="" v-model="text" id="" cols="30" rows="10" placeholder="share your experience"></textarea>
            <button type="submit">Submit</button>
        </form>
    </div>
</template>

<script>
import CloseButton from './CloseButton';
    export default {
        data() {
            return {
                text: ''
            }
        },
        methods: {
            submit() {
                axios.post('/api/posts', {text: this.$data.text}).then(()=> window.location.href = '/');
            }
        },
        components: {
            CloseButton,
        },
    }
</script>

<style lang="scss" scoped>
    @mixin modal_size{
        width: 100%;
        height: calc(100vh - 80px);
    }
    @mixin modal_vis{
        background: rgb(114, 93, 93);
        text-align: center;
    }
    .modal{
        @include modal_size();
        @include modal_vis();
    }
    @mixin cross-btn_size{
        font-size: 25px;
    }
    @mixin cross-btn_vis{
        background:honeydew;

    }
    @mixin cross-btn_pos{
        right: 0;
        position: absolute;
        top: 0;
    }
    .cross-btn{
        @include cross-btn_pos();
        @include cross_btn_size();
        @include cross-btn_vis();
    }
    @mixin text-body_size{
        width: calc(100% - 25px);
        height: calc(100% - 50px);
        margin-top: 30px;
        
    }
    .text-body{
        @include text-body_size();
    }
</style>
