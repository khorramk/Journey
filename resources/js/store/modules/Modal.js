import mutations from '../mutations';
import actions from '../actions';

const state = {
    open: false
}

export default {
    actions,
    mutations,
    state
}
