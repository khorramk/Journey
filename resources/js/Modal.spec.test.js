import { mount } from '@vue/test-utils';
import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import Modal from './components/Modal';
import expectExport from 'expect';


const wrapper = mount(Modal);

const vm = wrapper.vm;

console.log(wrapper);

it('should render', ()=> {
    expect(wrapper.element).toMatchSnapshot();
})
