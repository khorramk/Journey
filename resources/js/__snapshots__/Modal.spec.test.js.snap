// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  class="modal bg-blue-300 relative"
>
  <button
    class="cross-btn"
  >
    
        ( X )
    
  </button>
   
  <form>
    <textarea
      class="text-body"
      cols="30"
      id=""
      name=""
      placeholder="share your experience"
      rows="10"
    />
     
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
